[collection] Run sanity tests outside of our container

relrod opened the pull request:
##### SUMMARY

Also just ignore one sanity test for the export module, instead of ignoring all of them.

Also use latest ansible-test, and make it work on GHA (by using podman instead of docker).

Signed-off-by: Rick Elrod <rick@elrod.me>

<!--- Describe the change, including rationale and design decisions -->

Maybe a follow up to https://github.com/ansible/awx/pull/13356

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Collection

relrod committed:
Run sanity tests outside of our container

Also just ignore one sanity test for the export module, instead of
ignoring all of them.

Also use latest ansible-test, and make it work on GHA (by using podman
instead of docker).

Signed-off-by: Rick Elrod <rick@elrod.me>

relrod commented:
CC @felixfontein for feedback

gundalow commented on Makefile:
```diff
@@ -339,7 +339,7 @@ test_collection_sanity:
 	ansible --version
 	COLLECTION_VERSION=1.0.0 make install_collection
-	cd $(COLLECTION_INSTALL) && ansible-test sanity --exclude=plugins/modules/export.py
+	cd $(COLLECTION_INSTALL) && ansible-test sanity --docker
```
Nice `--docker` means we have the exact dependencies & Python versions installed as ansible-test requires.

AlanCoding replied:
One request I'd like to make - I need a local debugging cycle, and it might be hard to do that with this. If you don't mind, I'd like to add a new Makefile arg, like `SANITY_OPTIONS` that will default to `--docker`, but in the CLI we can clear them out. Because this does a lot of stuff that's good to have automated for the feedback cycle, like installing and `cd`ing into the right directory, which is very annoying manually.

While, yes, sometimes I would like to run --docker locally, that takes a long time, and a parameter can also narrow the test subset.

gundalow commented on awx_collection/tests/sanity/ignore-2.14.txt:
```diff
@@ -0,0 +1 @@
+plugins/modules/export.py validate-modules:nonexistent-parameter-documented  # needs awxkit to construct argspec
```
Nice
For others, this means that `ansible-test sanity` does run against plugins/modules/export.py though we ignore the errors about plugin parameters not matching documentation.

This is better than `--exclude=plugins/modules/export.py` as we still check for other errors.

AlanCoding commented on .github/workflows/ci.yml:
```diff
@@ -148,3 +145,21 @@ jobs:
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Upgrade ansible-core
```
Isn't this installing it in the first place?

shanemcd replied:
The Ubuntu GHA runner has Ansible installed by default.

gundalow commented on .github/workflows/ci.yml:
```diff
@@ -148,3 +145,21 @@ jobs:
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Upgrade ansible-core
```
```suggestion
      # The containers that GitHub Actions use have Ansible installed, so upgrade to make sure we have the latest version.
      - name: Upgrade ansible-core
```

AlanCoding committed:
Changes to run sanity tests locally

Use a Makefile arg for the ansible-test sanity CLI args
  defaults to --docker
  in the future we probably need to customize python versions

Copy the rule exception for Ansible 2.15
  this helps people who are running from Ansible devel

AlanCoding committed:
Add comment about Ansible-core being installed by default

Co-authored-by: John R Barker <john@johnrbarker.com>

AlanCoding commented:
Operator test failing due to Galaxy flake. Will try to resolve by closing and re-opening.
